### PR TITLE
Add noise uncorrelation (data pre-whitening) before the reconstruction

### PR DESCRIPTION
**Pull request by Laura2305**
- Related to discussion #135 
- Noise pre-whitening added in reconstruction_multiCoil

### CODE REVIEW

File: src/Reconstruction/RecoParameters.jl [hunk 1]
```diff
@@ -125,7 +125,7 @@ function setupIterativeReco(acqData::AcquisitionData{T}, recoParams::Dict) where
   # noise data acquisition [samples, coils]
   noiseData = get(recoParams, :noiseData, Complex{T}[])
   if isempty(noiseData)
-    L_inv = noiseData
+    L_inv = []
```
Comment (alexjaffray) [inline src/Reconstruction/RecoParameters.jl:128]: I think this change is good, explicitly initializing it as empty. Could also potentially initialize it as Nothing?

**Review by alexjaffray — COMMENTED** [review]
Great work! I've incorporated a few suggestions in my review. I think there is a question whether we should change the output of `setupIterativeReco` and the input arguments of `reconstruction_multiCoil` to accommodate the passing of `L_inv` or whether we should pass `L_inv` as part of the recoParams struct. Perhaps @tknopp can weigh in on this.

File: test/testReconstruction.jl [hunk 2]
```diff
@@ -357,6 +356,52 @@ function testSENSEReco(N = 64, T = ComplexF64)
...
+  noise_data = randn(Complex{Float64}, (N, div(N,2), numCoils)) .* 5
+  noise_data = reshape(noise_data, (N .* div(N,2), numCoils)) * psi
+  acqData.kdata[1,1,1] = acqData.kdata[1,1,1] .+ noise_data
+
+  # reco parameters
+  params = Dict{Symbol, Any}()
+  params[:reco] = "multiCoil" #"standard"
+  params[:reconSize] = (N,N)
+  params[:regularization] = "L2"
+  params[:iterations] = 100
```
[the remark this answers is not in the document]
Comment (Laura2305) [reply test/testReconstruction.jl]: 20 iterations are not enough to reach the solution, when I use 20 I can still see ghosting artefacts. I tried with 100 and the artefacts are still there, after the noise decorrelation. With 150 I get the desired result.

File: src/Tools/NoiseDeCorr.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        senseMapsUnCorr = reshape(senseMapsUnCorr, sizeMaps)
+    end
+
+    return senseMapsUnCorr
+end
+
+
+"""
+        covariance(noiseData)
+
```
Comment (JeffFessler) [inline src/Tools/NoiseDeCorr.jl:26]:
```suggestion
The `noiseData` array should be number of noise samples by number of coils.
```

docstring should clarify how `noiseData` is organized - perhaps # samples by # of coils?

**Comment by tknopp** [comment]
This PR looks very good, thank you @Laura2305. I would merge this if there are no objections.

**Comment by tknopp** [comment]
Let's merge this. The multi-echo case can be implemented later. I gave you access rights so that you can also work on branches within this repository. This is often easier than to use a fork.

**Pull request merged**